### PR TITLE
FOGL-7354 : Sqlite table insert optimisation

### PR DESCRIPTION
**Pull request by gnandan**
Signed-off-by: nandan <nandan.ghildiyal@gmail.com>

### CODE REVIEW

File: C/plugins/storage/sqlite/common/connection.cpp [hunk 1]
```diff
@@ -1189,9 +1189,11 @@ vector<string>  asset_codes;
  */
 int Connection::insert(const string& schema, const string& table, const string& data)
 {
-SQLBuffer	sql;
+string sqlCommand = {};
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlite/common/connection.cpp:1192]: Do not change away from using the SQLBuffer class, this was done explicitly for this purpose to improve performance by reducing string copies.

File: C/plugins/storage/sqlitelb/common/connection.cpp [hunk 2]
```diff
@@ -1265,17 +1265,16 @@ int Connection::insert(const std::string& schema,
 			const std::string& table,
 			const std::string& data)
 {
-SQLBuffer	sql;
+string sqlCommand = {};
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlitelb/common/connection.cpp:1268]: Please do not remove this use of SQLBuffer and take the backward step off using raw strings.

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]

File: C/plugins/storage/sqlitelb/common/connection.cpp (hunk 2 above)
Comment (gnandan) [reply C/plugins/storage/sqlitelb/common/connection.cpp]: fixed

File: C/plugins/storage/sqlite/common/connection.cpp (hunk 1 above)
Comment (gnandan) [reply C/plugins/storage/sqlite/common/connection.cpp]: fixed

**Review by Innovative-ashwin — DISMISSED** [review]

File: C/plugins/storage/sqlite/common/connection.cpp [hunk 3]
```diff
@@ -1226,13 +1227,11 @@ std::size_t arr = data.find("inserts");
...
 	}
 
-	// Start a trabsaction
-	sql.append("BEGIN TRANSACTION;");
-
 	// Number of inserts
 	int ins = 0;
-
-	// Iterate through insert array
+    int failedInsertCount = 0;
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlite/common/connection.cpp:1232]: Please maintain correct indentation

File: C/plugins/storage/sqlite/common/connection.cpp [hunk 4]
```diff
@@ -1244,138 +1243,153 @@ std::size_t arr = data.find("inserts");
...
+						sqlite3_bind_text(stmt, columID, SQLITE3_NOW, -1, SQLITE_TRANSIENT);
+					}
+					else
+					{	
+						sqlite3_bind_text(stmt, columID, escape(str).c_str(), -1, SQLITE_TRANSIENT);
+					}
 				}
-				else
+				else if (itr->value.IsDouble()) {
+					sqlite3_bind_double(stmt, columID,itr->value.IsDouble());
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlite/common/connection.cpp:1306]: sqlite3_bind_double(stmt, columID,itr->value.GetDouble());

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]

File: C/plugins/storage/sqlite/common/connection.cpp (hunk 3 above)
Comment (gnandan) [reply C/plugins/storage/sqlite/common/connection.cpp]: fixed

File: C/plugins/storage/sqlite/common/connection.cpp (hunk 4 above)
Comment (gnandan) [reply C/plugins/storage/sqlite/common/connection.cpp]: fixed

**Review by MarkRiddoch — APPROVED** [review]
This looks like it should be fine, but, given the sensitivity of this area, before merging to develop can you get the system test run against your branch please?

**Review by Mohit04tomar — APPROVED** [review]

**Pull request merged**